### PR TITLE
feat: add `contract.srcTxId` property

### PR DESCRIPTION
**Pull request by johnletey**
Allows contracts to view their own source txID.

### CODE REVIEW

**Review by hlolli — APPROVED** [review]
lgtm, bit unfortunate inconsistency in name-ing, but that was already there and not related to this PR.

**Comment by johnletey** [comment]
Yeah, completely agree @hlolli, can totally change it to `SmartWeave.contract.source`?

**Comment by hlolli** [comment]
it's fine @johnletey I think sourceId is better than source, for the intermediary variables, that can be refactored later.

File: src/contract-load.ts [hunk 1]
```diff
@@ -32,7 +32,13 @@ export async function loadContract(arweave: Arweave, contractID: string, contrac
     state = contractTX.get('data', { decode: true, string: true });
   }
 
-  const { handler, swGlobal } = createContractExecutionEnvironment(arweave, contractSrc, contractID, contractOwner);
+  const { handler, swGlobal } = createContractExecutionEnvironment(
+    arweave,
+    contractSrc,
+    contractID,
+    contractSrcTXID,
```
Comment (ppedziwiatr) [inline src/contract-load.ts:39]: IMO there's is some serious lack of naming consistency...
`contractSrcTXID`  vs `sourceId` vs `contractSourceId` 🤯  (and `contractID` without `TX` part...)

IMHO naming convention should be:
<subject>TxId

eg:
contractTxId
contractSrcTxId
interactionTxId

etc...

**Comment by hlolli** [comment]
Realizing that this change would require modifications into the documentation
at least here in the global api:
https://github.com/ArweaveTeam/SmartWeave/blob/master/CONTRACT-GUIDE.md

File: src/contract-load.ts (hunk 1 above)
Comment (hlolli) [reply src/contract-load.ts:39]: In relation to this PR, I'd say your suggestion of `contractSrcTxId` makes sense. The other variable name changes would fit another PR.

**Comment by johnletey** [comment]
@ppedziwiatr @hlolli How does it look now? To me personally seems a bit off as now we have `contract.contractBlah`.

**Comment by cedriking** [comment]
I think having stuff like: `contract.contractSomething` is a bad way of naming things.
Looking at the code, right now, seems that everywhere it was called `contract.id`, so this one should just be `contract.srcId`?

**Comment by ppedziwiatr** [comment]
> I think having stuff like: `contract.contractSomething` is a bad nomenclature.
> Looking at the code, right now, seems that everywhere it was called `contract.id`, so this one should just be `contract.srcId`?

Since everything is going to be re-written from scratch soon (I hope ;-)), it's probably not that important...I personally would prefer `contract.srcTxId` in case if it is nested property, and sth. like `contractSrcTxId` for "lone" variables - like in this example - https://github.com/ArweaveTeam/SmartWeave/pull/84#pullrequestreview-695591727
But that's just personal perference, not that important ;-)

**Comment by cedriking** [comment]
> > I think having stuff like: `contract.contractSomething` is a bad nomenclature.
> > Looking at the code, right now, seems that everywhere it was called `contract.id`, so this one should just be `contract.srcId`?
> 
> Since everything is going to be re-written from scratch soon (I hope ;-)), it's probably not that important...I personally would prefer `contract.srcTxId` in case if it is nested property, and sth. like `contractSrcTxId` for "lone" variables - like in this example - [#84 (review)](https://github.com/ArweaveTeam/SmartWeave/pull/84#pullrequestreview-695591727)
> But that's just personal perference, not that important ;-)

If rewriten, I would say `contract.srcTxId` and `contract.txId` would be the best approach (to me).

**Comment by johnletey** [comment]
Okay, folks, I just updated it to `contract.srcTxId` 😄

Ready to merge? @hlolli @cedriking

**Pull request closed without merging**